pkg/activator/handler/tracing_handler_test.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handler
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/logging"
+	rtesting "knative.dev/pkg/reconciler/testing"
+	"knative.dev/pkg/tracing"
+	"knative.dev/pkg/tracing/config"
+	tracetesting "knative.dev/pkg/tracing/testing"
+	activatorconfig "knative.dev/serving/pkg/activator/config"
+)
+
+func TestTracingHandler(t *testing.T) {
+	tests := []struct {
+		name           string
+		tracingEnabled bool
+	}{{
+		name:           "enabled",
+		tracingEnabled: true,
+	}, {
+		name:           "disabled",
+		tracingEnabled: false,
+	}}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctx, cancel, _ := rtesting.SetupFakeContextWithCancel(t)
+			defer cancel()
+
+			baseHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+			handler := NewTracingHandler(baseHandler)
+
+			resp := httptest.NewRecorder()
+			req := httptest.NewRequest(http.MethodPost, "http://example.com", nil)
+			const traceID = "821e0d50d931235a5ba3fa42eddddd8f"
```

Comment:
this is very specific :)

Reply:
It is, but the thing wants something that looks like an actual ID. "foo" doesn't work :joy: